package/MDAnalysis/analysis/gnm.py

```diff
@@ -234,46 +262,27 @@ def generate_kirchoff(self):
         Returns
         -------
         array
-             the resulting Kirchhoff matrix
+                the resulting Kirchhoff matrix
         """
         positions = self.ca.positions
 
-        natoms = len(positions)
-
         #add the com from each bonus group to the ca_positions list
         for item in self.Bonus_groups:
             #bonus = self.u.select_atoms(item)
             positions = np.vstack((positions, item.center_of_mass()))
-            natoms += 1
 
-        matrix = np.zeros((natoms, natoms), "float")
-        res_positions, grid, low_x, low_y, low_z = generate_grid(positions, self.cutoff)
-        icounter = 0
+        natoms = len(positions)
+        matrix = np.zeros((natoms, natoms), np.float64)
 
         cutoffsq = self.cutoff ** 2
 
-        for icounter in range(natoms):
-            #find neighbours from the grid
-            neighbour_atoms = []
-            for x in (-1, 0, 1):
-                #print icounter, natoms, len(positions), len(res_positions)
-                gx = res_positions[icounter][0] + x
-                if 0 <= gx < len(grid):
-                    for y in (-1, 0, 1):
-                        gy = res_positions[icounter][1] + y
-                        if 0 <= gy < len(grid[0]):
-                            for z in (-1, 0, 1):
-                                gz = res_positions[icounter][2] + z
-                                if 0 <= gz < len(grid[0][0]):
-                                    neighbour_atoms += grid[gx][gy][gz]
-
-            #for jcounter in range(icounter+1,natoms):
-            for jcounter in neighbour_atoms:
-                if jcounter > icounter and _dsq(positions[icounter], positions[jcounter]) < cutoffsq:
-                    matrix[icounter][jcounter] = -1.0
-                    matrix[jcounter][icounter] = -1.0
-                    matrix[icounter][icounter] = matrix[icounter][icounter] + 1
-                    matrix[jcounter][jcounter] = matrix[jcounter][jcounter] + 1
+        for i_atom, j_atom in neighbour_generator(positions, self.cutoff):
+            if j_atom > i_atom and _dsq(positions[i_atom], positions[j_atom]) < cutoffsq:
```

Comment:
I'm still not sure if this should be in the `neighbor_generator` function.  The `_dsq` call is quite expensive and including it might allow further optimizations.